programs/monaco_protocol/src/instructions/matching/matching_pool.rs

```diff
@@ -3,40 +3,17 @@ use anchor_lang::prelude::*;
 use crate::state::market_account::{Market, MarketStatus};
 use crate::state::market_matching_pool_account::MarketMatchingPool;
 use crate::state::market_matching_queue_account::MarketMatchingQueue;
-use crate::state::market_outcome_account::MarketOutcome;
 use crate::{CoreError, Order};
 
 pub fn update_on_match(
-    market_outcome: &mut Account<MarketOutcome>,
     market_matching_pool_against: &mut Account<MarketMatchingPool>,
     market_matching_pool_for: &mut Account<MarketMatchingPool>,
-    market_account: &Pubkey,
     stake_matched: u64,
     for_order: &Account<Order>,
     against_order: &Account<Order>,
 ) -> Result<()> {
     let for_fully_matched = for_order.stake_unmatched == 0_u64;
     let against_fully_matched = against_order.stake_unmatched == 0_u64;
-    require!(
-        market_outcome.market.eq(market_account),
-        CoreError::MarketDoesNotMatch
-    );
-
-    // market-outcome stats
-    msg!("market: calculating market-outcome stats");
-    if stake_matched > 0_u64 {
-        market_outcome.latest_matched_price =
-            if for_order.creation_timestamp < against_order.creation_timestamp {
-                for_order.expected_price
-            } else {
-                against_order.expected_price
-            };
-
-        market_outcome.matched_total = market_outcome
-            .matched_total
-            .checked_add(stake_matched)
-            .ok_or(CoreError::MatchingMatchedAmountUpdateError)?;
-    }
```

Comment:
this should not be here in a first place; moved level up

Reply:
Maybe a little pointless, since the only codepath to here is via `match_orders` which is probably up for deletion entirely now for v0.15.0, but sure, no problem